flowapi/flowapi/user_model.py

```diff
@@ -88,24 +93,39 @@ def _get_query_kind_and_aggregation_unit(self, query_json: dict) -> Tuple[str, s
 
         Returns
         -------
-        str, str
-            Query kind and aggregation unit
+        list of tuples str, str
+            List of tuples giving query kind and aggregation unit
 
         """
 
         try:
             query_kind = query_json["query_kind"]
+        except KeyError:
+            raise MissingQueryKindError
+        try:
             if query_kind == "spatial_aggregate":
-                return self._get_query_kind_and_aggregation_unit(
+                return self._get_query_kinds_and_aggregation_units(
                     query_json=query_json["locations"]
                 )
-        except KeyError:
-            raise MissingQueryKindError
+            elif query_kind == "joined_spatial_aggregate":
```

Comment:
Just as a note for future refactorings: the appearance of multiple if/else branches and special case treatments for different query kinds suggests to me that this is not the right place for this logic to live. Will give it some thought, but it should likely be each individual query's responsibility to know which types of permissions it needs and to verify these against given claims.